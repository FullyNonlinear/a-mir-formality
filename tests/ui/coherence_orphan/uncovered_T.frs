[
    crate core {
        trait CoreTrait<ty T> {}
    },
    crate foo {
        struct FooStruct {}
        impl<ty T> CoreTrait<FooStruct> for T {}
    }
]
