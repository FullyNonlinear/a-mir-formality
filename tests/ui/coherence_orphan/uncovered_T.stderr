Error: orphan_check(impl <ty> CoreTrait <FooStruct> for ^ty0_0 { })

Caused by:
    judgment `prove_wc_list { goal: {@ IsLocal(CoreTrait(!ty_0, FooStruct))}, assumptions: {}, env: Env { variables: [!ty_0], coherence_mode: true }, decls: decls(222, [trait CoreTrait <ty, ty> ], [impl <ty> CoreTrait(^ty0_0, FooStruct)], [], [], [], [adt FooStruct ], {}, {FooStruct}) }` failed at the following rule(s):
      the rule "some" failed at step #0 (crates/formality-prove/src/prove/prove_wc_list.rs:28:14) because
        judgment `prove_wc { goal: @ IsLocal(CoreTrait(!ty_0, FooStruct)), assumptions: {}, env: Env { variables: [!ty_0], coherence_mode: true }, decls: decls(222, [trait CoreTrait <ty, ty> ], [impl <ty> CoreTrait(^ty0_0, FooStruct)], [], [], [], [adt FooStruct ], {}, {FooStruct}) }` failed at the following rule(s):
          the rule "trait ref is local" failed at step #0 (crates/formality-prove/src/prove/prove_wc.rs:110:14) because
            judgment `is_local_trait_ref { goal: CoreTrait(!ty_0, FooStruct), assumptions: {}, env: Env { variables: [!ty_0], coherence_mode: true }, decls: decls(222, [trait CoreTrait <ty, ty> ], [impl <ty> CoreTrait(^ty0_0, FooStruct)], [], [], [], [adt FooStruct ], {}, {FooStruct}) }` failed at the following rule(s):
              the rule "local parameter" failed at step #1 (crates/formality-prove/src/prove/is_local.rs:77:14) because
                judgment `is_local_parameter { goal: !ty_0, assumptions: {}, env: Env { variables: [!ty_0], coherence_mode: true }, decls: decls(222, [trait CoreTrait <ty, ty> ], [impl <ty> CoreTrait(^ty0_0, FooStruct)], [], [], [], [adt FooStruct ], {}, {FooStruct}) }` failed at the following rule(s):
                  the rule "local parameter" failed at step #0 (crates/formality-prove/src/prove/is_local.rs:138:14) because
                    judgment `prove_normalize { p: !ty_0, assumptions: {}, env: Env { variables: [!ty_0], coherence_mode: false }, decls: decls(222, [trait CoreTrait <ty, ty> ], [impl <ty> CoreTrait(^ty0_0, FooStruct)], [], [], [], [adt FooStruct ], {}, {FooStruct}) }` failed at the following rule(s):
                      the rule "normalize-via-assumption" failed at step #0 (crates/formality-prove/src/prove/prove_normalize.rs:26:14) because
                        expression evaluated to an empty collection: `&assumptions`
              the rule "local trait" failed at step #0 (crates/formality-prove/src/prove/is_local.rs:70:17) because
                condition evaluted to false: `decls.is_local_trait_id(&goal.trait_id)`
    
